src/Assistant.ts

```diff
@@ -30,14 +28,16 @@ export interface AssistantConfig {
  * Callback utilities
  */
 interface AssistantUtilityArgs {
-  getThreadContext: GetThreadContextFn;
-  saveThreadContext: SaveThreadContextFn;
+  getThreadContext: GetThreadContextUtilFn;
+  saveThreadContext: SaveThreadContextUtilFn;
   say: SayFn;
   setStatus: SetStatusFn;
   setSuggestedPrompts: SetSuggestedPromptsFn;
   setTitle: SetTitleFn;
 }
 
+type GetThreadContextUtilFn = () => Promise<AssistantThreadContext>;
```

Comment:
Despite the ongoing issue with non-ack'd events, it appears to work as expected. This update corresponds to the enrich function. I believe I've gotten the pass-through correct:

```
// Do not pass preparedArgs (ie, do not add utilities to get/save)
preparedArgs.getThreadContext = () => threadContextStore.get(args);
preparedArgs.saveThreadContext = () => threadContextStore.save(args);
```